flood_adapt/dbs_classes/dbs_template.py

```diff
@@ -0,0 +1,370 @@
+from flood_adapt.object_model.interface.events import IEvent
+from flood_adapt.object_model.interface.measures import IMeasure
+from flood_adapt.object_model.interface.projections import IProjection
+from flood_adapt.object_model.interface.scenarios import IScenario
+from flood_adapt.object_model.interface.strategies import IStrategy
+
+ObjectModel = Union[IScenario, IEvent, IProjection, IStrategy, IMeasure, IBenefit]
+
+
+class DbsTemplate(AbstractDatabaseElement):
+    _type = ""
+    _folder_name = ""
+    _object_model_class = None
+
+    def __init__(self, database: IDatabase):
+        """
+        Initialize any necessary attributes.
+        """
+        self.input_path = database.input_path
+        self._path = self.input_path / self._folder_name
+        self._database = database
+
+    def get(self, name: str) -> ObjectModel:
+        """Returns an object of the type of the database with the given name.
+
+        Parameters
+        ----------
+        name : str
+            name of the object to be returned
+
+        Returns
+        -------
+        ObjectModel
+            object of the type of the specified object model
+        """
+        # Make the full path to the object
+        full_path = self._path / name / f"{name}.toml"
+
+        # Check if the object exists
+        if not Path(full_path).is_file():
+            raise ValueError(f"{self._type.capitalize()} '{name}' does not exist.")
+
+        # Load and return the object
+        object_model = self._object_model_class.load_file(full_path)
+        return object_model
+
+    def set_lock(self, model_object: ObjectModel = None, name: str = None) -> None:
+        """Locks the element in the database to prevent other processes from accessing it. The object can be locked by
+        providing either the model_object or the name. If both are provided, the model_object is used. An element can
+        be locked multiple times. For example, if 2 scenario's are running that use the same event, it should be locked
+        twice. The lock is only released when both scenario's are finished.
+
+        Parameters
+        ----------
+        model_object : ObjectModel, optional
+            The model_object to lock, by default None
+        name : str, optional
+            The name of the model_object to lock, by default None
+
+        Raises
+        ------
+        ValueError
+            Raise error if both model_object and name are None.
+        """
+        # If both model_object and name are None, raise error
+        if model_object is None and name is None:
+            raise ValueError("Either model_object or name must be provided.")
+
+        # If only name is provided, get the model_object
+        if model_object is None:
+            model_object = self.get(name)
+
+        # Set the lock and save the object
+        model_object.attrs.lock_count += 1
+        self.save(model_object, overwrite=True)
```

Comment:
I think there might be a bug here.
If we start with an obj that is already locked, and want to use it in another higher level obj:
self.save(obj, overwrite=True) gets called, which calls self.delete(obj)
which then will raise a ValueError at L259